include/picongpu/particles/externalBeam/StartAttributesImpl.hpp

```diff
@@ -0,0 +1,203 @@
+                     * position.
+                     *
+                     * @tparam T_Particle type of the particles that should be created
+                     *
+                     * @param realParticlesPerCell number of new real particles in the cell in which this instance
+                     * creates particles
+                     *
+                     * @return number of macro particles that need to be created in this cell (The operator() will be
+                     * called that many times)
+                     */
+                    template<typename T_Particle>
+                    DINLINE uint32_t numberOfMacroParticles(float_X const realParticlesPerCell)
+                    {
+                        // numberOfMacroParticles also initializes the start position functor
+                        const uint32_t numMacroParticles
+                            = T_AccStartPositionFunctor::template numberOfMacroParticles<T_Particle>(
+                                context_m,
+                                realParticlesPerCell);
+
+                        return numMacroParticles;
+                    }
+
+                private:
+                    PMACC_ALIGN(context_m, T_Context);
+                };
+            } // namespace acc
+
+
+            //! Host factory implementation for StartAttributes
+            template<
+                typename T_Species,
+                typename T_StartPositionFunctor,
+                typename T_MomentumFunctor,
+                typename... T_Functors>
+            struct StartAttributesImpl
+                : private T_StartPositionFunctor
+                , private T_MomentumFunctor
+                , private T_Functors...
+
+            {
+                using RNGFactory = pmacc::random::RNGProvider<simDim, random::Generator>;
+
+                HINLINE StartAttributesImpl(uint32_t const& currentStep)
+                    : T_StartPositionFunctor(currentStep)
+                    , T_MomentumFunctor(currentStep)
+                    , T_Functors(currentStep)...
+                    , rngHandle(RNGFactory::createHandle())
+                {
+                }
+
+                /** Create functor for the accelerator
+                 *
+                 * @tparam T_Worker lockstep worker type
+                 *
+                 * @param worker lockstep worker
+                 * @param localSupercellOffset offset (in superCells, without any guards) relative
+                 *                        to the origin of the local domain
+                 */
+                template<typename T_Worker>
+                DINLINE auto operator()(T_Worker const& worker, DataSpace<simDim> const& localSupercellOffset) const
+                {
+                    // initialize a random nuber generator for the sub-functors
+                    RNGFactory::Handle rngHandleLocal = rngHandle;
+                    rngHandleLocal.init(
```

Comment:
please have a look here https://github.com/ComputationalRadiationPhysics/picongpu/pull/4685/files#diff-f8b4116fa9a761b713c81fdcca550f38c0af63541ee3c9c81853285d17cdc868R148 and https://github.com/ComputationalRadiationPhysics/picongpu/pull/4685#issue-1900792592

The way RNG numbers are mapped to worker threads has changed. SImple copy-past fix to this PR is required.